app/controllers/authentication.js

```diff
@@ -116,185 +117,207 @@ var Controller = function(){
+        redirect_uri: req.body.redirectUri
+      };
+
+      request.get({url: accessTokenUrl, qs: params},
+        function(err, response, accessToken)
+        {
+          if (err) { callback(err); return; }
+
+          accessToken = qs.parse(accessToken);
+          var headers = { 'User-Agent': 'Satellizer' };
+
+          callback(null, accessToken, headers);
+        });
     };
 
-    // Exchange authorization code for access token.
-    request.get({ url: accessTokenUrl,
-                  qs: params
-                }, 
-                function(err, response, accessToken) 
-                {
-                  accessToken = qs.parse(accessToken);
-                  var headers = { 'User-Agent': 'Satellizer' };
-
-      // Retrieve profile information about the current user.
-      request.get({ url: userApiUrl,
-                    qs: accessToken,
-                    headers: headers,
-                    json: true
-                  }, function(err, response, profile) {
-
-        if(response.statusCode != 200) 
+    var getProfile = function(accessToken, headers, callback)
+    {
+      request.get({url: userApiUrl, qs: accessToken, headers: headers, json: true}, 
+        function(err, response, profile)
         {
-          return res.status(409).json({
-            message: profile.message
-          });
-        }
+          if (err) { callback(err); return; }
+
+          if(response.statusCode != 200) 
+          {
+            return res.status(409).json({
+              message: profile.message
+            });
+          }
+
+          if(!profile.email)
+          {
+            return res.status(409).json({
+              message: 'Did not get email address.'
+            });
+          }
+
+          callback(null, accessToken, headers, profile);
+        });
+    };
 
-        if(!profile.email)
+    var checkOrganization = function(accessToken, headers, profile, callback) 
+    {
+      request.get({url: orgsApiUrl, qs: accessToken, headers: headers, json: true},
+        function(err, response)
         {
-          return res.status(409).json({
-            message: 'Did not get email address.'
+          if (err) { callback(err); return; }
+
+          var belongsOrg = _.find(response.body, {id: parseInt(nconf.get('github').orgId)});
+
+          if (belongsOrg)
+          {
+            callback(null, accessToken, headers, profile);
+          }
+          else
+          {
+            return res.status(401).send({
+              message: 'You do not have required access.'
+            })
+          }
+        });
+    };
+
+    var checkAdmin = function(accessToken, headers, profile, callback)
+    {
+      request.get({ url: teamsApiUrl, qs: accessToken, headers: headers, json: true},
+        function(err, response)
+        {
+          if (err) { callback(err); return; }
+
+          var isAdmin = _.find(response.body, {id: parseInt(nconf.get('github').adminTeamId)});
```

Comment:
I think it is better to use name instead of id because name is human readable... reason why you selected id here ?